satellite/satellitedb/overlaycache.go

```diff
@@ -22,6 +24,235 @@ type overlaycache struct {
+	}()
+
+	newNodes, err := sqlRowsToNodes(rows)
+	if err != nil {
+		return nil, err
+	}
+
+	return newNodes, nil
+}
+
+func sqlRowsToNodes(rows *sql.Rows) (nodes []*pb.Node, err error) {
+	for rows.Next() {
+		overlayNode := &dbx.OverlayCacheNode{}
+		err = rows.Scan(&overlayNode.NodeId, &overlayNode.NodeType,
+			&overlayNode.Address, &overlayNode.FreeBandwidth, &overlayNode.FreeDisk,
+			&overlayNode.AuditSuccessRatio, &overlayNode.AuditUptimeRatio,
+			&overlayNode.AuditCount, &overlayNode.AuditSuccessCount,
+			&overlayNode.UptimeCount, &overlayNode.UptimeSuccessCount)
+		if err != nil {
+			return nil, err
+		}
+
+		node, err := convertOverlayNode(overlayNode)
+		if err != nil {
+			return nil, err
+		}
+		nodes = append(nodes, node)
+	}
+	return nodes, nil
+}
+
+func (cache *overlaycache) findReputableNodesQuery(ctx context.Context, req *getNodesRequest) (*sql.Rows, error) {
+	auditCount := req.minReputation.AuditCount
+	auditSuccessRatio := req.minReputation.AuditSuccessRatio
+	uptimeCount := req.minReputation.UptimeCount
+	uptimeRatio := req.minReputation.UptimeRatio
+	nodeAmt := req.reputableNodeAmount
+
+	var args []interface{}
+	var rows *sql.Rows
+	var err error
+	var nodeTypeStorage int32 = 2
+
+	if len(req.excluded) == 0 {
+		args = append(args, auditCount, req.newNodeAuditThreshold,
+			auditSuccessRatio, uptimeCount, uptimeRatio,
+			req.freeBandwidth, req.freeDisk, nodeTypeStorage, nodeAmt)
+
+		// This queries for nodes whose audit counts are greater than or equal to
+		// the new node audit threshold and the minimum reputation audit count.
+		rows, err = cache.db.Query(cache.db.Rebind(`SELECT node_id,
+	node_type, address, free_bandwidth, free_disk, audit_success_ratio,
+	audit_uptime_ratio, audit_count, audit_success_count, uptime_count,
+	uptime_success_count
+	FROM overlay_cache_nodes
+	WHERE audit_count >= ?
+	AND audit_count >= ?
+	AND audit_success_ratio > ?
+	AND uptime_count > ?
+	AND audit_uptime_ratio > ?
+	AND free_bandwidth > ?
+	AND free_disk > ?
+	AND node_type == ?
+	LIMIT ?
+	`),
+			args...)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		args = append(args, auditCount, req.newNodeAuditThreshold,
+			auditSuccessRatio, uptimeCount, uptimeRatio,
+			req.freeBandwidth, req.freeDisk, pq.Array(req.excluded), nodeTypeStorage, nodeAmt)
+
+		// This queries for nodes whose audit counts are greater than or equal to
+		// the new node audit threshold and the minimum reputation audit count.
+		rows, err = cache.db.Query(cache.db.Rebind(`SELECT node_id,
+	node_type, address, free_bandwidth, free_disk, audit_success_ratio,
+	audit_uptime_ratio, audit_count, audit_success_count, uptime_count,
+	uptime_success_count
+	FROM overlay_cache_nodes
+	WHERE audit_count >= ?
+	AND audit_count >= ?
+	AND audit_success_ratio > ?
+	AND uptime_count > ?
+	AND audit_uptime_ratio > ?
+	AND free_bandwidth > ?
+	AND free_disk > ?
+	AND node_id NOT IN ?::text[]
+	AND node_type == ?
+	LIMIT ?
```

Comment:
It looks like these two query cases can probably be merged now (since an empty excluded list shouldn't cause a problem anymore). Does that seem right?

Reply:
yeah that makes sense!